js/ui/interaction.js

```diff
@@ -1,325 +0,0 @@
-    },
-
-    _onMouseDown: function (e) {
-        this._map.stop();
-        this._startPos = DOM.mousePos(this._el, e);
-        this._fireMouseEvent('mousedown', e);
-    },
-
-    _onMouseUp: function (e) {
-        var map = this._map,
-            rotating = map.dragRotate && map.dragRotate.isActive();
-
-        if (this._contextMenuEvent && !rotating) {
-            this._fireMouseEvent('contextmenu', this._contextMenuEvent);
-        }
-
-        this._contextMenuEvent = null;
-        this._fireMouseEvent('mouseup', e);
-    },
-
-    _onMouseMove: function (e) {
-        var map = this._map,
-            el = this._el;
-
-        if (map.dragPan && map.dragPan.isActive()) return;
-        if (map.dragRotate && map.dragRotate.isActive()) return;
-
-        var target = e.toElement || e.target;
-        while (target && target !== el) target = target.parentNode;
-        if (target !== el) return;
-
-        this._fireMouseEvent('mousemove', e);
-    },
-
-    _onTouchStart: function (e) {
-        this._map.stop();
-        this._fireTouchEvent('touchstart', e);
-
-        if (!e.touches || e.touches.length > 1) return;
-
-        if (!this._tapped) {
-            this._tapped = setTimeout(this._onTouchTimeout, 300);
-
-        } else {
-            clearTimeout(this._tapped);
-            this._tapped = null;
-            this._fireMouseEvent('dblclick', e);
-        }
-    },
-
-    _onTouchMove: function (e) {
-        this._fireTouchEvent('touchmove', e);
-    },
-
-    _onTouchEnd: function (e) {
-        this._fireTouchEvent('touchend', e);
-    },
-
-    _onTouchCancel: function (e) {
-        this._fireTouchEvent('touchcancel', e);
-    },
-
-    _onTouchTimeout: function () {
-        this._tapped = null;
-    },
-
-    _onClick: function (e) {
-        var pos = DOM.mousePos(this._el, e);
-
-        if (pos.equals(this._startPos)) {
-            this._fireMouseEvent('click', e);
-        }
-    },
-
-    _onDblClick: function (e) {
-        this._fireMouseEvent('dblclick', e);
-        e.preventDefault();
-    },
-
-    _onContextMenu: function (e) {
-        this._contextMenuEvent = e;
-        e.preventDefault();
-    },
-
-    _fireMouseEvent: function (type, e) {
-        var pos = DOM.mousePos(this._el, e);
-
-        return this._map.fire(type, {
-            lngLat: this._map.unproject(pos),
-            point: pos,
-            originalEvent: e
-        });
-    },
-
-    _fireTouchEvent: function (type, e) {
-        var touches = DOM.touchPos(this._el, e),
-            singular = touches.reduce(function (prev, curr, i, arr) {
-                return prev.add(curr.div(arr.length));
-            }, new Point(0, 0));
-
-        return this._map.fire(type, {
-            lngLat: this._map.unproject(singular),
-            point: singular,
-            lngLats: touches.map(function(t) { return this._map.unproject(t); }, this),
-            points: touches,
-            originalEvent: e
-        });
-    }
-};
-
-
-/**
```

Comment:
I moved it to `Map`! Given that the events are fired on `Map`, that feels like the right place for these docs.